### PR TITLE
(#605) Unbreak bolt_task with multiple files

### PR DESCRIPTION
**Pull request by smortex**
Allow running a Bolt task that use extra files beside the task script itself.

### CODE REVIEW

**Comment by smortex** [comment]
Ping @jay7x!  This is a follow-up to #605

Using this patch, I can run [a Bolt tasks](https://github.com/opus-codium/puppet-odoo/blob/master/tasks/databases_list.sh) that needs [an extra file](https://github.com/opus-codium/puppet-odoo/blob/master/files/databases-list.py).

I do not use much more than this for now, so if you can test how it behave in your use case and report feedback here, it would be awesome!

**Comment by smortex** [comment]
For the record, in order to test, I stop puppet and overwrite the files in `/opt/puppetlabs/mcollective/plugins/mcollective/` with the ones from this repository.  When I am done, I restart puppet which restores the original files.

File: spec/unit/mcollective/util/tasks_support_spec.rb [hunk 1]
```diff
@@ -361,21 +363,35 @@ module Util
...
         end
       end
 
+      describe "#populate_spooldir" do
+        let(:spooldir) do
+          "/tmp/tasks-spool-#{$$}"
+        end
+        it "should copy files" do
+          FileUtils.stubs(:mkdir_p).with(File.join(spooldir, "files"), :mode => 0o750)
+          FileUtils.stubs(:cp).with(File.join(cache, "f3b4821836cf7fe6fe17dfb2924ff6897eba43a44cc4cba0e0ed136b27934ede"), File.join(spooldir, "files", "ls.rb"))
```
Comment (ripienaar) [inline spec/unit/mcollective/util/tasks_support_spec.rb:382]: better make these `expects` I think, then it actually checks the calls are done

**Review by ripienaar — CHANGES_REQUESTED** [review]

File: spec/unit/mcollective/util/tasks_support_spec.rb (hunk 1 above)
Comment (ripienaar) [reply spec/unit/mcollective/util/tasks_support_spec.rb:382]: looks amazing thank you, just that one thing

Comment (smortex) [reply spec/unit/mcollective/util/tasks_support_spec.rb]: Good catch, thanks!

**Comment by ripienaar** [comment]
looks good to me

**Review by ripienaar — APPROVED** [review]

**Comment by ripienaar** [comment]
you all ready with this? I'll go ahead and merge then

**Comment by smortex** [comment]
That looks ok for me, my end-to-end testing is currently limited by #439 which I started to work on at $WORK (but and authentication failure by the database is exactly what I would expect from my code running as the wrong user).

The only "inconsistency" I see when comparing with Bolt is that the task script is in `<spooldir>/script` while with Bolt it's in `<tempdir>/<modulename>/tasks/script`.  But I do not think this is a problem since it's the "entry point" and nothing should refer to this file.

**Pull request merged**